src/app/users/store/users.reducer.ts

```diff
@@ -1,41 +1,51 @@
 import { IUser } from '../user.interface';
 import { createReducer, on, Action } from '@ngrx/store';
 import { getAllUsersActions } from './users.actions';
+import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';
+import { state } from '@angular/animations';
 
+//export interface IUsersInitialState extends EntityState<IUser> {
 export interface IUsersInitialState {
   users: IUser[];
-  getAllUsersPending: boolean;
```

Comment:
and why did you remove this?

Reply:
I cannot use the same interface for the rest of the reducers because it will have an unproper attribute. 
GetAllUsers is only applicable for the get all users action.
Please let me know.

Reply:
Wrong concept,  you must use the same interface of users Feature initial state.
this pattern should be used for this interface.
```getAllUsersPending
getAllUserseError
updateUsersPending
updateUsersError
getUsersPending
getUsersError
createUsersPending
createUsersError
```

Reply:
I did an aproch similar to a video that I have been following. Please let me know If it is ok, otherwise I change it as you say. Thanks